Source/Core/Core/Debugger/Debugger_SymbolMap.cpp

```diff
@@ -100,52 +100,52 @@ bool GetCallstack(std::vector<CallstackEntry>& output)
 
 void PrintCallstack()
 {
-  printf("== STACK TRACE - SP = %08x ==", PowerPC::ppcState.gpr[1]);
+  fmt::print("== STACK TRACE - SP = {:08x} ==", PowerPC::ppcState.gpr[1]);
 
   if (LR == 0)
   {
-    printf(" LR = 0 - this is bad");
+    fmt::print(" LR = 0 - this is bad");
   }
 
   if (g_symbolDB.GetDescription(PC) != g_symbolDB.GetDescription(LR))
   {
-    printf(" * %s  [ LR = %08x ]", g_symbolDB.GetDescription(LR).c_str(), LR);
+    fmt::print(" * {}  [ LR = {:08x} ]", g_symbolDB.GetDescription(LR), LR);
   }
 
   WalkTheStack([](u32 func_addr) {
     std::string func_desc = g_symbolDB.GetDescription(func_addr);
     if (func_desc.empty() || func_desc == "Invalid")
       func_desc = "(unknown)";
-    printf(" * %s [ addr = %08x ]", func_desc.c_str(), func_addr);
+    fmt::print(" * {} [ addr = {:08x} ]", func_desc, func_addr);
   });
 }
 
 void PrintCallstack(Common::Log::LOG_TYPE type, Common::Log::LOG_LEVELS level)
 {
-  GENERIC_LOG(type, level, "== STACK TRACE - SP = %08x ==", PowerPC::ppcState.gpr[1]);
+  GENERIC_LOG_FMT(type, level, "== STACK TRACE - SP = {:08x} ==", PowerPC::ppcState.gpr[1]);
 
   if (LR == 0)
   {
-    GENERIC_LOG(type, level, " LR = 0 - this is bad");
+    GENERIC_LOG_FMT(type, level, " LR = 0 - this is bad");
   }
 
   if (g_symbolDB.GetDescription(PC) != g_symbolDB.GetDescription(LR))
   {
-    GENERIC_LOG(type, level, " * %s  [ LR = %08x ]", g_symbolDB.GetDescription(LR).c_str(), LR);
+    GENERIC_LOG_FMT(type, level, " * {}  [ LR = {:08x} ]", g_symbolDB.GetDescription(LR), LR);
   }
 
   WalkTheStack([type, level](u32 func_addr) {
     std::string func_desc = g_symbolDB.GetDescription(func_addr);
     if (func_desc.empty() || func_desc == "Invalid")
       func_desc = "(unknown)";
-    GENERIC_LOG(type, level, " * %s [ addr = %08x ]", func_desc.c_str(), func_addr);
+    GENERIC_LOG_FMT(type, level, " * {} [ addr = {:08x} ]", func_desc, func_addr);
   });
 }
 
 void PrintDataBuffer(Common::Log::LOG_TYPE type, const u8* data, size_t size,
                      const std::string& title)
 {
-  GENERIC_LOG(type, Common::Log::LDEBUG, "%s", title.c_str());
+  GENERIC_LOG_FMT(type, Common::Log::LDEBUG, "{}", title);
```

Comment:
Any reason why we'd want to keep this as `GENERIC_LOG_FMT` with a fixed `Log` enum value, vs. calling `DEBUG_LOG_FMT` instead? On one hand, the other methods in here pass a level and use that; but on the other this method doesn't have this level parameter at all. I can see the "consistency" argument looming (which I'd usually bring to the table as well), but this kinda looks weird at the moment.

Thoughts?

Reply:
This is necessary, since the regular debug log (and other individual level logs handle the type parameter like:

```cpp
#define DEBUG_LOG_FMT(t, ...)                                                                      \
  do                                                                                               \
  {                                                                                                \
    GENERIC_LOG_FMT(Common::Log::t, Common::Log::LDEBUG, __VA_ARGS__);                             \
  } while (0)
```

vs

```cpp
#define GENERIC_LOG_FMT(t, v, ...)                                                                 \
  do                                                                                               \
  {                                                                                                \
    if (v <= MAX_LOGLEVEL)                                                                         \
      Common::Log::GenericLogFmt(v, t, __FILE__, __LINE__, __VA_ARGS__);                           \
  } while (0)
```

so it's not possible to pass that through as is, since as far as the macro cares:

```cpp
DEBUG_LOG_FMT(type, "{}", title);
```

would result in the qualifier:

```cpp
Common::Log::type
```

being formed, which doesn't exist.

Reply:
Oh, I completely missed that one. Don't mind me then :D